slatedb/src/utils.rs

```diff
@@ -655,4 +782,48 @@ mod tests {
         let result = timeout_future.await;
         assert_eq!(result.unwrap(), 42);
     }
+
```

Comment:
we should add some reader tests as well

Reply:
done, and used rstest for both reader/writer tests so that they're consistent. note that the proptesting for the serde in the other seqtracker encoding tests this code path as well pretty heavily